test/integration/install.py

```diff
@@ -26,7 +28,12 @@ def __init__(self, path):
 					(bitness_py, bitness_driver))
 
 	def _install_driver_win(self, ephemeral):
-		with psutil.Popen(["msiexec.exe", "/i", self._driver_path, "/norestart", "/quiet"]) as p:
+		(fd, log_name) = tempfile.mkstemp(suffix="-installer.log")
+		os.close(fd)
+		if ephemeral:
+			atexit.register(shutil.rmtree, log_name, ignore_errors=True)
+
+		with psutil.Popen(["msiexec.exe", "/i", self._driver_path, "/norestart", "/quiet", "/lx", log_name]) as p:
```

Comment:
Does `/lx` _only_ give you the “extended” information? The docs in https://docs.microsoft.com/en-us/windows/desktop/msi/command-line-options suggest that you need to specify other options too, maybe `/l*x`?

Reply:
The extended one should already be relevant and is more concise (below an example), but having everything is probably better indeed, so I've extended the logging.

```
=== Logging started: 12/02/2019  09:41:04 ===
Action start 09:41:04: INSTALL.
Action start 09:41:04: FindRelatedProducts.
Action ended 09:41:04: FindRelatedProducts. Return value 1.
Action start 09:41:04: AppSearch.
Action ended 09:41:04: AppSearch. Return value 1.
Action start 09:41:04: LaunchConditions.
Action ended 09:41:04: LaunchConditions. Return value 1.
Action start 09:41:04: ValidateProductID.
Action ended 09:41:04: ValidateProductID. Return value 1.
MSI (s) (E4:40) [09:41:04:269]: Baseline: Sorting baselines for {E87C5D53-FDDF-4539-9447-4903AED527BC}.
MSI (s) (E4:40) [09:41:04:269]: Baseline: New baseline 8.0.0 from transaction.
MSI (s) (E4:40) [09:41:04:270]: Baseline: Sorted order Native: Order 0.
MSI (s) (E4:40) [09:41:04:270]: Baseline Data Table:
MSI (s) (E4:40) [09:41:04:270]: ProductCode: {E87C5D53-FDDF-4539-9447-4903AED527BC} Version: 8.0.0 Attributes: 0 PatchId: Native BaselineId: -2147483648 Order: 0
MSI (s) (E4:40) [09:41:04:270]: Baseline File Table:
MSI (s) (E4:40) [09:41:04:271]: Delta compression fallback method for this product transaction is 'MSI 2.0 legacy obsolescence'
Action start 09:41:04: CostInitialize.
Action ended 09:41:04: CostInitialize. Return value 1.
Action start 09:41:04: FileCost.
Action ended 09:41:04: FileCost. Return value 1.
Action start 09:41:04: CostFinalize.
Action ended 09:41:04: CostFinalize. Return value 1.
Action start 09:41:04: SetODBCFolders.
Action ended 09:41:04: SetODBCFolders. Return value 0.
Action start 09:41:04: MigrateFeatureStates.
Action ended 09:41:04: MigrateFeatureStates. Return value 0.
Action start 09:41:04: InstallValidate.
Action ended 09:41:04: InstallValidate. Return value 1.
Action start 09:41:04: RemoveExistingProducts.
Action ended 09:41:04: RemoveExistingProducts. Return value 1.
Action start 09:41:04: InstallInitialize.
Action ended 09:41:04: InstallInitialize. Return value 1.
MSI (s) (E4:40) [09:41:04:344]: SECREPAIR: Hash Value for the file:esodbc-8.0.0-SNAPSHOT-windows-x86_64.msi is: 65v3slA2GyRNyuXUZKbdafISdxswXFo5GBTdKmyzxFQ=
Action start 09:41:04: ProcessComponents.
Action ended 09:41:04: ProcessComponents. Return value 1.
Action start 09:41:04: UnpublishFeatures.
Action ended 09:41:04: UnpublishFeatures. Return value 1.
Action start 09:41:04: RemoveODBC.
Action ended 09:41:04: RemoveODBC. Return value 1.
Action start 09:41:04: RemoveFiles.
Action ended 09:41:04: RemoveFiles. Return value 0.
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'esdsnbnd.dll_2079851965' has no eligible binary patches
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'esdsnedt.dll_841069281' has no eligible binary patches
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'esodbc8u.lib_1979525861' has no eligible binary patches
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'LICENSE.rtf_38926593' has no eligible binary patches
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'LICENSE.txt_412198531' has no eligible binary patches
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'NOTICE.txt_381543480' has no eligible binary patches
MSI (s) (E4:40) [09:41:04:357]: The file represented by File table key 'esodbc8u.dll_41842506' has no eligible binary patches
Action start 09:41:04: InstallFiles.
Action ended 09:41:04: InstallFiles. Return value 1.
Action start 09:41:04: InstallODBC.
Action ended 09:41:04: InstallODBC. Return value 0.
Action start 09:41:04: RegisterUser.
Action ended 09:41:04: RegisterUser. Return value 1.
Action start 09:41:04: RegisterProduct.
Action ended 09:41:04: RegisterProduct. Return value 1.
Action start 09:41:04: PublishFeatures.
Action ended 09:41:04: PublishFeatures. Return value 1.
Action start 09:41:04: PublishProduct.
Action ended 09:41:04: PublishProduct. Return value 1.
Action start 09:41:04: InstallFinalize.
MSI (s) (E4:40) [09:41:04:376]: Product: Elasticsearch ODBC Driver (64bit) -- Error 1925. You do not have sufficient privileges to complete this installation for all users of the machine. Log on as administrator and then retry this installation.

Action ended 09:41:04: InstallFinalize. Return value 3.
Action ended 09:41:04: INSTALL. Return value 3.
MSI (s) (E4:40) [09:41:04:393]: Product: Elasticsearch ODBC Driver (64bit) -- Installation failed.

MSI (s) (E4:40) [09:41:04:393]: Windows Installer installed the product. Product Name: Elasticsearch ODBC Driver (64bit). Product Version: 8.0.0.0. Product Language: 1033. Manufacturer: Elasticsearch B.V.. Installation success or error status: 1603.

MSI (s) (E4:40) [09:41:04:393]: Closing MSIHANDLE (1) of type 790542 for thread 18240
=== Logging stopped: 12/02/2019  09:41:04 ===
```